native/screens/SignUp.jsx

```diff
@@ -1,180 +1,309 @@
-//             onChange={(e) => handleChange(e, "email", setNewUser)}
-//             type="email"
-//             value={newUser["email"]}
-//             onFocus={() => setEmailError(null)}
-//             onBlur={(e) => handleEmailCheck(e)}
-//           />
-//         </div>
-//         <div className="form-error">
-//           <h6>{emailError}</h6>
-//         </div>
-//         <div className="input-wrapper">
-//           <label htmlFor="password">Password</label>
-//           <input
-//             required
-//             id="password"
-//             name="password"
-//             onChange={(e) => handleChange(e, "password", setNewUser)}
-//             type="password"
-//             value={newUser["password"]}
-//           />
-//         </div>
-//         <div className="input-wrapper">
-//           <label htmlFor="confirm_password">Re-enter Password</label>
-//           <input
-//             required
-//             id="confirm_password"
-//             name="confirm_password"
-//             onChange={(e) => handleChange(e, "confirm_password", setNewUser)}
-//             type="password"
-//             value={newUser["confirm_password"]}
-//           />
-//         </div>
-//         <div className="form-error">
-//           {newUser.password !== "" && (
-//             <h6>
-//               {newUser.password === newUser.confirm_password
-//                 ? "Passwords match."
-//                 : "Passwords do not match."}
-//             </h6>
-//           )}
-//         </div>
-//         <SingleActionButton text="Register" type="submit" />
-//       </form>
-//       <h6>
-//         Already have an account? <Link to="/sign-in">Sign in.</Link>
-//       </h6>
-//     </div>
-//   );
-// };
+  return (
+    <View style={styles.accountForms}>
+      <Modal
+        visible={successModalVisible}
+        transparent={true}
+        animationType="slide"
+        >
+        <View style={styles.modalContainer}>
+          <Text style={styles.centeredView}>Success!</Text>
+            {/* Replace these with SingleButton or DoubleButton component*/}
+          <TouchableOpacity
+            style={styles.singleButton}
+            onPress={()=>handleReroute('EditProfile')}
+            color="white">
+            <Text style={styles.buttonText}>Edit Profile</Text>
+          </TouchableOpacity>
+          <TouchableOpacity
+            style={styles.singleButton}
+            onPress={()=> handleReroute('Roulette')}
+            color="white">
+            <Text style={styles.buttonText}>Go to Roulette</Text>
+          </TouchableOpacity>
+        </View>
+      </Modal>
+      <Text style={styles.title} >Create an Account</Text>
+      {/* Replace with Form componenet? */}
+      <View style={styles.inputContainer}>
+        <Text>Name</Text>
+        <TextInput 
+          value={newUser.first_name}
+          style={styles.input}
+          onChangeText={(firstName) => setNewUser((prevState) => {
```

Comment:
take a look at the handler in `./services/utils/handlers` to see if that function can work for you. otherwise, create a generic function in this component to handle the onChangeText and DRY up the logic